com.chartboost.mediation/Android~/chartboost-mediation-android-bridge/src/main/java/com/chartboost/mediation/unity/BannerLayout.kt

```diff
@@ -4,15 +4,26 @@ import android.content.Context
 import android.view.MotionEvent
 import android.widget.RelativeLayout
 import com.chartboost.heliumsdk.ad.HeliumBannerAd
-import kotlin.math.sqrt
 import kotlin.math.pow
+import kotlin.math.sqrt
 
-class BannerLayout(
-    context: Context,
-    private var bannerView: HeliumBannerAd,
+class BannerLayout : RelativeLayout {
+    private var bannerView: HeliumBannerAd
     private var dragListener: IBannerDragListener
-) : RelativeLayout(context) {
-    var canDrag: Boolean = true
+    
+    constructor(
+        context: Context,
+        bannerView: HeliumBannerAd,
+        dragListener: IBannerDragListener
+    ) : super(context) {
+        this.bannerView = bannerView
+        this.dragListener = dragListener
+
+        // making it clickable here allows onInterceptTouchEvent to intercept touch events on bannerView
+        bannerView.isClickable = true;
```

Comment:
Am curious... will this still trigger our click callbacks? 🤔

Reply:
Yeah they work. I tested it on device. The touch/click on banner is intercepted by it's parent layout which checks if it is a drag. If no drag then it passes on the touch to its children views which will trigger click callbacks